specs/capri/DNS_Resolution_Proposal_in_Hotpot.md

```diff
@@ -0,0 +1,238 @@
+	LogFlushFrequency  time.Duration `conf:"log_flush_frequency,5s"` // Default value is 5s
+	HTTPSEnabled       bool          `conf:"https_enabled,false"`
+	BeegoHTTPSCertFile string        `conf:"beego_https_cert_file,/opt/opensds-security/opensds/opensds-cert.pem"`
+	BeegoHTTPSKeyFile  string        `conf:"beego_https_key_file,/opt/opensds-security/opensds/opensds-key.pem"`
+}
+
+type OsdsLet struct {
+	ApiEndpoint       string        `conf:"api_endpoint,localhost:50049"`
++	DnsEndpoint       string        `conf:"dns_endpoint,localhost:50049"`
+	Daemon            bool          `conf:"daemon,false"`
+	LogFlushFrequency time.Duration `conf:"log_flush_frequency,5s"` // Default value is 5s
+}
+
+type OsdsDock struct {
+	ApiEndpoint                string        `conf:"api_endpoint,localhost:50050"`
++	DnsEndpoint                string        `conf:"dns_endpoint,localhost:50050"`
+	DockType                   string        `conf:"dock_type,provisioner"`
+	EnabledBackends            []string      `conf:"enabled_backends,lvm"`
+	Daemon                     bool          `conf:"daemon,false"`
+	BindIp                     string        `conf:"bind_ip"` // Just used for attacher dock
+	HostBasedReplicationDriver string        `conf:"host_based_replication_driver,drbd"`
+	LogFlushFrequency          time.Duration `conf:"log_flush_frequency,5s"` // Default value is 5s
+	Backends
+}
+
+type Database struct {
+-	Credential string `conf:"credential,username:password@tcp(ip:port)/dbname"`
+-	Driver     string `conf:"driver,etcd"`
+-	Endpoint   string `conf:"endpoint,localhost:2379,localhost:2380"`
++	Credential  string `conf:"credential,username:password@tcp(ip:port)/dbname"`
++	Driver      string `conf:"driver,etcd"`
++	DnsEndpoint string `conf:"dns_endpoint,localhost:2379,localhost:2380"`
+}
+```
+
+### REST API impact
+
+None
+
+### Security impact
+
+None
+
+### Other end user impact
+
+None
+
+### Performance impact
+
+None
+
+### Other deployer impact
+
+To enable DNS resolution, users need to change the `opensds.conf` like below:
+```conf
+[osdsapiserver]
+api_endpoint = 0.0.0.0:50040
++dns_endpoint = localhost:50040
+auth_strategy = keystone
+# If https is enabled, the default value of cert file
+# is /opt/opensds-security/opensds/opensds-cert.pem,
+# and key file is /opt/opensds-security/opensds/opensds-key.pem
+https_enabled = False
+beego_https_cert_file =
+beego_https_key_file =
+# Encryption and decryption tool. Default value is aes.
+password_decrypt_tool = aes
+
+[osdslet]
+api_endpoint = 0.0.0.0:50049
++dns_endpoint = localhost:50049
+
+[osdsdock]
+api_endpoint = 0.0.0.0:50050
++dns_endpoint = localhost:50050
+# Choose the type of dock resource, only support 'provisioner' and 'attacher'.
+dock_type = provisioner
+# Specify which backends should be enabled, sample,ceph,cinder,lvm and so on.
+enabled_backends = sample
+
+[database]
+-credential = opensds:password@127.0.0.1:3306/dbname
```

Comment:
What is replacing the credential?

Reply:
credential is only required for mysql database, so leave it here seems a bit confusing.